### PR TITLE
Issue 3114 - watch should return cached data immediately

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://github.com/openshift/origin/issues/3114

### CODE REVIEW

**Comment by jwforres** [comment]
Fixes https://github.com/openshift/origin/issues/3114

File: assets/app/scripts/services/data.js [hunk 1]
```diff
@@ -376,15 +376,24 @@ angular.module('openshiftConsole')
       this._watchOptions(type, context, opts);
     }
 
+    var existingData;
     if (this._watchInFlight(type, context) && this._resourceVersion(type, context)) {
       if (callback) {
         callback(this._data(type, context));
       }
     }
     else if (this._listInFlight(type, context)) {
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: refactor to only check existing data once?
```
if (watchinflight) {
}
else {
  ... existing data check and callback ...
  if !listopinflight {
    ...start list op...
  }
}
```

Comment (liggitt) [inline assets/app/scripts/services/data.js]: only check for existing data if we have a callback

**Comment by jwforres** [comment]
@liggitt updated

File: assets/app/scripts/services/data.js [hunk 2]
```diff
@@ -381,11 +381,16 @@ angular.module('openshiftConsole')
         callback(this._data(type, context));
       }
     }
-    else if (this._listInFlight(type, context)) {
-      // no-op, our callback will get called when listOperation completes
-    }
     else {
-      this._startListOp(type, context);
+      if (callback) {
+        var existingData = this._data(type, context);
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: jslint will probably complain about the nested var declaration

**Comment by liggitt** [comment]
LGTM if jslint is happy

**Comment by jwforres** [comment]
[merge]

**Pull request merged**